cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/TemplatesClassloaderUtil.java

```diff
@@ -0,0 +1,351 @@
+        Files.walkFileTree(templateRoot, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.toString().endsWith(".class")) {
+                    foundPaths.add(file);
+                    LOG.debug("    * Found class file {}", file);
+                }
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                // Log errors but do not throw an exception
+                LOG.warn("An IOException occurred while reading file on path {} with message: {}", file,
+                    exc.getMessage());
+                LOG.debug("An IOException occurred while reading file on path {} with message: {}", file,
+                    LOG.isDebugEnabled() ? exc : null);
+                return FileVisitResult.CONTINUE;
+            }
+        });
+        return foundPaths;
+    }
+
+    /**
+     * Walks the jar file in search of utility classes
+     *
+     * @param jarUri
+     *            URI of jar file
+     * @param jarfs
+     *            FileSystem of jar file
+     * @return List<String> of file paths containing class files
+     * @throws IOException
+     *             if file could not be visited
+     */
+    private static List<String> walkJarFile(URI jarUri, FileSystem jarfs) throws IOException {
+        List<String> foundClasses = new LinkedList<>();
+        // walk the jar file
+        LOG.debug("Searching for classes in {}", jarUri);
+        Files.walkFileTree(jarfs.getPath("/"), new SimpleFileVisitor<Path>() {
+
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (file.toString().endsWith(".class")) {
+                    LOG.debug("    * Found class file {}", file);
+                    // remove the leading '/' and the trailing '.class'
+                    String fileName = file.toString().substring(1, file.toString().length() - 6);
+                    // replace the path separator '/' with package separator '.' and add it to the
+                    // list of found files
+                    foundClasses.add(fileName.replace("/", "."));
+                }
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                // Log errors but do not throw an exception
+                LOG.warn("An IOException occurred while reading file on path {} with message: {}", file,
+                    exc.getMessage());
+                LOG.debug("An IOException occurred while reading file on path {} with message: {}", file,
+                    LOG.isDebugEnabled() ? exc : null);
```

Comment:
You can put this last parameter simply in the warn log before

Reply:
adjusted.